docs/advanced.md

```diff
@@ -52,6 +52,21 @@ finally:
     client.close()
 ```
 
+There is some overhead to managing these connections, so you should try to persist your client as much as possible.
+For example, in an ASGI web application like Starlette or FastAPI you should create the client in the lifespan event so that it gets re-used for each request.
+If you know what servers you are going to be connecting to it is also good practice to create a client instance per server.
+This maximizes connection re-use and minimizes connection turnover.
+
+For example, if you are connecting to `http://auth.example.com` and `http://service.example.com` you would create two distinct client instances:
+
+```python
+with httpx.Client(base_url="http://auth.example.com") as auth_client:
+    with httpx.Client(base_url="http://service.example.com") as service_client:
+        resp = auth_client.get("/.well-known/jwks.json")        
+        ...
+        resp = service_client.get("/endpoint")
+```
```

Comment:
This is an interesting point. Do we want two distinct client instances here? The connection pooling is smart enough to retain connections across each domain wherever possible.